resource/resource_graph.go

```diff
@@ -45,28 +45,18 @@ func (g *Graph) CurrLogicalClockValue() int64 {
 	return g.logicalClock.Load()
 }
 
-func (g *Graph) getAllChildrenOf(node Name) graphNodes {
-	if _, ok := g.nodes[node]; !ok {
-		return nil
-	}
-	out := graphNodes{}
-	for k, parents := range g.parents {
-		if _, ok := parents[node]; ok {
-			out[k] = &GraphNode{}
```

Comment:
The function signature of returning the actual children nodes (instead of just the names) made sense. We clearly don't do that. I'm concerned someone in the future will expect this to be a real graph node. And I'm concerned that returning the "real" graph node here is unsafe under concurrent operations.

Nothing was actually using these graph nodes, just the names. So I figured I could punt on understanding these hypotheticals and get safety by just removing the unsafe part of the API.

Reply:
> I'm concerned someone in the future will expect this to be a real graph node. 

A fair concern, as I recently expected the values of the returned map to be actual graph nodes. I think the new API makes more sense.